teamcity: failed tests on master: testrace/TestLogic, testrace/TestLogic/distsql, testrace/TestLogic/distsql/storing

The following tests appear to have failed:

[#317614](https://teamcity.cockroachdb.com/viewLog.html?buildId=317614):

```
--- FAIL: testrace/TestLogic (574.770s)
<autogenerated>:10: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic372729944
<autogenerated>:9: use -show-logs to present logs inline
--- FAIL: testrace/TestLogic/distsql (2.000s)
null
--- FAIL: testrace/TestLogic/distsql/storing (31.360s)
logic_test.go:1723: 
	testdata/logic_test/storing:203: expected success, but found
	pq: index "t14601a" does not exist
logic_test.go:1734: 
	testdata/logic_test/storing:203: error in statement, skipping to next file

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/table.go
+++ b/pkg/sql/table.go
@@ -784,11 +784,19 @@ func (p *planner) expandIndexName(
 	}
 
 	if index.SearchTable {
-		realTableName, err := p.findTableContainingIndex(ctx, p.txn, p.getVirtualTabler(), tn.DatabaseName, tn.TableName)
+		// On the first call to expandIndexName(), index.Index is empty and
+		// index.Table is the index name. Once the table name is resolved
+		// for the index below, index.Table references a new table name
+		// (not the index), so a subsequent call to expandIndexName()
+		// will generate tn using the new value of index.Table, which
+		// is a table name. Therefore assign index.Index only once.
+		if index.Index == "" {
+			index.Index = tn.TableName
+		}
+		realTableName, err := p.findTableContainingIndex(ctx, p.txn, p.getVirtualTabler(), tn.DatabaseName, index.Index)
 		if err != nil {
 			return nil, err
 		}
-		index.Index = tn.TableName
 		index.Table.TableNameReference = realTableName
 		tn = realTableName
 	}

